### PR TITLE
[DO NOT MERGE] Thinking in loopback

### PR DESCRIPTION
**Pull request by bajtos**
As part of my work on https://github.com/strongloop/loopback-next/wiki/Thinking-in-LoopBack, I am writing real code to verify the code snippets in the wiki actually works. This branch contains the code I wrote to support the wiki page.

Note that I skipped schema definitions via helper functions, because it was easier for me to simply drop the full Swagger spec document.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,9 +33,12 @@
     "@loopback/authentication": "^4.0.0-alpha.5",
     "@loopback/context": "^4.0.0-alpha.11",
     "@loopback/core": "^4.0.0-alpha.11",
+    "@loopback/repository": "^4.0.0-alpha.7.tgz",
```
Comment (raymondfeng) [inline package.json]: Why `^4.0.0-alpha.7.tgz`?

File: src/application.ts [hunk 2]
```diff
@@ -8,15 +8,21 @@ import {validateApiSpec} from '@loopback/testlab'
...
     super({
       components: [AuthenticationComponent],
       sequence: MySequence,
     });
+    this.api(spec);
     this.bind(AuthenticationBindings.STRATEGY)
       .toProvider(MyAuthStrategyProvider);
     this.controller(HelloWorldController);
+    this.controller(ProductController);
+    this.bind('repositories.Product').toClass(ProductRepository);
```
Comment (raymondfeng) [inline src/application.ts]: To be replaced by `this.repository()` when it's merged in.

File: package.json (hunk 1 above)
Comment (bajtos) [reply package.json]: Ah, I was installing locally from a file and incorrectly changed the version spec for this commit.

File: src/application.ts (hunk 2 above)
Comment (bajtos) [reply src/application.ts]: Yes!

File: src/controllers/product.controller.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ import {inject} from '@loopback/context';
 import {EntityCrudRepository} from '@loopback/repository';
 import {ProductRepository} from '../repositories/product.repository';
 import {Product} from '../models/product.model';
+import {HttpErrors} from '@loopback/core';
```
Comment (Setogit) [inline src/controllers/product.controller.ts]: sort the import items?

**Pull request closed without merging**